java-manta-client/src/main/java/com/joyent/manta/client/multipart/MantaMultipartManager.java

```diff
@@ -0,0 +1,1106 @@
+     *
+     * @return stream of objects representing files being uploaded via multipart
+     * @throws IOException thrown when there are network issues
+     */
+    public Stream<MantaMultipartUpload> listInProgress() throws IOException {
+        final List<Exception> exceptions = new ArrayList<>();
+
+        /* This nesting structure is unfortunate, but an artifact of us needing
+         * to close the stream when we have finished processing. */
+        try (Stream<MantaObject> multipartDirList = mantaClient
+                    .listObjects(this.resolvedMultipartUploadDirectory)
+                    .filter(MantaObject::isDirectory)) {
+
+            final Stream<MantaMultipartUpload> stream = multipartDirList
+                    .map(object -> {
+                        String idString = MantaUtils.lastItemInPath(object.getPath());
+                        UUID id = UUID.fromString(idString);
+
+                        try {
+                            MultipartMetadata mantaMetadata = downloadMultipartMetadata(id);
+                            return new MantaMultipartUpload(id, mantaMetadata.getPath());
+                        } catch (MantaClientHttpResponseException e) {
+                            if (e.getStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                                return null;
+                            } else {
+                                exceptions.add(e);
+                                return null;
+                            }
+                        } catch (IOException | RuntimeException e) {
+                            exceptions.add(e);
+                            return null;
+                        }
+                    })
+                    /* We explicitly filter out items that stopped existing when we
+                     * went to get the multipart metadata because we encountered a
+                     * race condition. */
+                    .filter(value -> value != null);
+
+            if (exceptions.isEmpty()) {
+                return stream;
+            }
+        // This catches an exception on the initial listObjects call
+        } catch (MantaClientHttpResponseException e) {
+            if (e.getStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                return Stream.empty();
+            } else {
+                throw e;
+            }
+        }
+
+        final MantaIOException aggregateException = new MantaIOException(
+                "Problem(s) listing multipart uploads in progress");
+
+        int count = 1;
+        for (Exception e : exceptions) {
+            final String label = String.format("exception_%d", count++);
+            aggregateException.setContextValue(label, e);
+        }
+
+        throw aggregateException;
+    }
+
+    /**
+     * Initializes a new multipart upload for an object.
+     *
+     * @param path remote path of Manta object to be uploaded
+     * @return unique id for the multipart upload
```

Comment:
nit: Makes it sound like the return type is a uuid instead of `MantaMultipartUpload`?